### PR TITLE
Pyats install script

### PR DESCRIPTION
**Pull request by nyu058**

### CODE REVIEW

File: pyats_installer.py [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+        'uninstall': ['pyats[full]']
+    },
+    '19.11':{
+        'install': ['unicon', 'unicon.plugins'],
+        'uninstall': ['unicon']
+    }
+}
+# todo find a way to automatically find pkg deps+extras
+# pkg_resource.require doesnt work b/c it raise exception if the dep is not installed
+PYATS_PKG_DEPENDENCIES = {
```
Comment (simingy) [inline pyats_installer.py]: look into https://github.com/naiquevin/pipdeptree/blob/master/pipdeptree.py

FrozenRequirement is what you need - I think

File: pyats_installer.py [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+    def __init__(self, extra, version=None, uninstall=None, **kwargs):
+        self.uninstall = uninstall
+        self.extra = extra
+        self.version = version or LATEST
+        # genie deps changed after 20.1
+        self.GENIE_PKG_DEPENDENCIES = NEW_GENIE_PKG_DEPENDENCIES if StrictVersion(
+            self.version) >= StrictVersion('20.1') else OLD_GENIE_PKG_DEPENDENCIES
+        
+        try:
+            import pyats.aetest
```
Comment (simingy) [inline pyats_installer.py]: i think the best way to go about this is to run a pip uninstall command.

Comment (nyu058) [reply pyats_installer.py]: I'm not sure about how pip uninstall command could be used here....

File: pyats_installer.py [hunk 3]
```diff
@@ -0,0 +1,229 @@
+#!/usr/bin/python
+import subprocess
+import argparse
+from distutils.version import StrictVersion
+LATEST = '20.1'
+
+VERSION_MAPPING = {
```
[the remark this answers is not in the document]
Comment (nyu058) [reply pyats_installer.py]: fixed

File: pyats_installer.py [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+
+                if self.downgrade or not self.latest:
+                    install_cmd = self._get_install_cmd()
+                else:
+                    install_cmd = self._get_install_cmd(True)
+
+                cmd = ';'.join([self._get_uninstall_cmd(), install_cmd])
+
+        print(cmd)
+        p = subprocess.Popen(cmd, stdout=subprocess.PIPE, shell=True)
```
[the remark this answers is not in the document]
Comment (nyu058) [reply pyats_installer.py]: done

File: pyats_installer.py [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+        for line in p.stdout:
+            print(line.decode(), end='')
```
[the remark this answers is not in the document]
Comment (nyu058) [reply pyats_installer.py]: done

File: pyats_installer.py [hunk 6]
```diff
@@ -0,0 +1,229 @@
...
+    'pyats.kleenex',
+    'pyats.connections',
+    'pyats.datastructures',
+    'pyats.easypy',
+    'pyats.results',
+    'pyats.reporter',
+    'pyats.tcl',
+    'pyats.topology',
+    'pyats.utils',
+    'pyats.templates'}
```
[the remark this answers is not in the document]
Comment (nyu058) [reply pyats_installer.py]: added

File: pyats_installer.py [hunk 7]
```diff
@@ -0,0 +1,229 @@
+#!/usr/bin/python
```
[the remark this answers is not in the document]
Comment (nyu058) [reply pyats_installer.py]: done

File: README.md [hunk 8]
```diff
@@ -4,6 +4,30 @@ This is a place-holder repository used only so there's a bug tracking/forum/wiki
...
 Use the issue tab above to create issues for pyATS.
 
+This repository also contains a pyats_installer.py script to help install/uninstall/upgrade/downgrade pyats versions. 
+Please consult below section on how to use this script.
+
+#### Using the pyATS installer script
+First make sure you have python installed, and either clone this repository or curl the script file:
+
+```
+curl -OL https://raw.githubusercontent.com/nyu058/pyats/master/pyats_installer.py
```
[the remark this answers is not in the document]
Comment (nyu058) [reply README.md]: fixed

**Comment by simingy** [comment]
no longer required and now part of `pyats version update` command

**Pull request closed without merging**